data/src/gadm/preprocess_gadm.py

```diff
@@ -7,150 +7,193 @@
-    # Drop the rows with missing country names
-    merged_gdf = merged_gdf.dropna(subset=["country"])
-    # Save the merged data
-    merged_gdf.to_file("../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson", driver="GeoJSON")
-    print("Merged data saved at: ../../raw_data/gadm/gadm_410-levels_ADM_0_merged.geojson")
+def prepare_adm_datasets(adm0: gpd.GeoDataFrame, adm1: gpd.GeoDataFrame) -> gpd.GeoDataFrame:
+    """
+    Merge the two datasets on the 'GID_0' and 'GID_1' columns. concatenating countries with Macao & Hongkong
+    """
+    adm1_prep = (
+        adm1[adm1["NAME_1"].isin(["Hong Kong", "Macau"])][["GID_1", "NAME_1", "geometry"]]
+        .rename(columns={"GID_1": "country_code", "NAME_1": "country"})
+        .pipe(lambda df: df.assign(country_code=df.country_code.str.replace("CHN.", "")))
+    )
 
-    return merged_gdf
+    return pd.concat(
+        [
+            adm0.rename(columns={"COUNTRY": "country", "GID_0": "country_code"}),
+            adm1_prep,
+        ],
+        ignore_index=True,
+    )[["geometry", "country_code"]]
 
 
 # 4. Generate SQL INSERT statement
-def generate_sql_insert(row):
+def generate_row_sql(row: pd.Series) -> str:
     sql_template = """
-    INSERT INTO public.countries
-    (code, name, continent, region_1, region_2, numeric_code, hdi, geometry)
-    VALUES ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', ''{region_2}'', {numeric_code}, {hdi}, '{geometry}');
-    """  # noqa: E501
-
-    geometry_wkt = row["geometry"].wkt  # Get WKT representation of the geometry
-    sql = sql_template.format(
+    ('{code}', '{name}', '{continent}'::public.countries_continent_enum, '{region_1}', ''{region_2}'', {numeric_code}, {hdi}, ST_GeomFromWKB(decode({geometry}, 'hex'), 4326), {area_ha})
+    """
+    geometry_wkb = row["geometry"].wkb.hex()  # Get WKb representation of the geometry
+    sql_row = sql_template.format(
         code=row["country_code"],
         name=row["country"],
         continent=row["continent_id"],
         region_1=row["region_1"] if pd.notnull(row["region_1"]) else "NULL",
         region_2=row["region_2"] if pd.notnull(row["region_2"]) else "NULL",
         numeric_code=row["numeric"],
         hdi=row["hdi_id"] if pd.notnull(row["hdi_id"]) else "NULL",
-        geometry=geometry_wkt,
+        geometry=geometry_wkb,
+        area_ha=row["area_ha"],
     )
-    return sql
+    return sql_row
+
+
+def generate_sql_insert(rows: str) -> str:
+    return f"""
+    INSERT INTO public.countries
+    (code, name, continent, region_1, region_2, numeric_code, hdi, geometry, area_ha)
+    VALUES
+    {rows};
+    ON CONFLICT(code)
+    DO UPDATE SET
+    area_ha = EXCLUDED.area_ha,
+    geometry = EXCLUDED.geometry;
+    """
 
 
 # 5. Write SQL to a file
-def write_sql_file(merged_gdf, output_file):
+def write_sql_file(merged_gdf: gpd.GeoDataFrame, output_file: str) -> None:
```

Comment:
Where is this file being saved? Can't see any path reference so I assume it is writing it in the current path

However the new file is under `/data/excel`. Is this the correct location?

Reply:
the sql statement is, the csv in in the raw_data/gadm folder.